app/src/main/java/org/schabi/newpipe/settings/NotificationsSettingsFragment.kt

```diff
@@ -0,0 +1,108 @@
+package org.schabi.newpipe.settings
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.content.SharedPreferences.OnSharedPreferenceChangeListener
+import android.os.Bundle
+import android.view.View
+import androidx.core.content.ContextCompat
+import com.google.android.material.snackbar.Snackbar
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import org.schabi.newpipe.BuildConfig
+import org.schabi.newpipe.R
+import org.schabi.newpipe.database.subscription.NotificationMode
+import org.schabi.newpipe.database.subscription.SubscriptionEntity
+import org.schabi.newpipe.local.subscription.SubscriptionManager
+import org.schabi.newpipe.notifications.NotificationHelper
+import org.schabi.newpipe.notifications.NotificationWorker
+import org.schabi.newpipe.notifications.ScheduleOptions
+import java.util.*
+
+class NotificationsSettingsFragment : BasePreferenceFragment(), OnSharedPreferenceChangeListener {
+
+    private var notificationWarningSnackbar: Snackbar? = null
+    private var loader: Disposable? = null
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        addPreferencesFromResource(R.xml.notifications_settings)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        defaultPreferences.registerOnSharedPreferenceChangeListener(this)
+    }
+
+    override fun onStop() {
+        defaultPreferences.unregisterOnSharedPreferenceChangeListener(this)
+        super.onStop()
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
+        val ctx = context ?: return
```

Comment:
This usage of `ctx` looks a bit strange, though I don't know if this piece of code can be reworded